projects/shr-mediator/src/botswana/services/terminology.service.ts

```diff
@@ -201,51 +205,25 @@ export class TerminologyService {
     }
   }
 
-  async getMappedCode(q: string): Promise<any> {
+  getMappedCodeFromMemory(mappings: any[], code: string): any {
     try {
-      const codeMapping = await this.getOclMapping(q);
-
-      //this.logger.log(`Code Mapping: ${JSON.stringify(codeMapping)}`)
+      this.logger.log(`Code: ${code}`)
+      const mapping = mappings.find((x: any) => x.from_concept_code === code);
 
-      if (codeMapping && codeMapping.length > 0) {
+      if (mapping) {
         return {
-          code: codeMapping[0].to_concept_code,
-          display: codeMapping[0].to_concept_name_resolved,
+          code: mapping.to_concept_code,
+          display: mapping.to_concept_name_resolved,
         };
       } else {
         return {};
```

Comment:
If mapping is not found, let's throw and exception or return undefined or null

Reply:
Fixed